src/main/java/org/sunbird/operationalreports/service/OperationalReportServiceImpl.java

```diff
@@ -73,82 +76,96 @@ public class OperationalReportServiceImpl implements OperationalReportService {
-                List<Map<String, Object>> contents = (List<Map<String, Object>>) userSearchResponse.get(Constants.CONTENT);
-                Set<String> rootOrgIds = new HashSet<>();
-                String rootOrgId = null;
-                if (!CollectionUtils.isEmpty(contents)) {
-                    for (Map<String, Object> content : contents) {
-                        rootOrgId = (String) content.get(Constants.ROOT_ORG_ID);
-                        rootOrgIds.add(rootOrgId);
-                    }
+        while (true) {
+            try {
+                Map<String, Object> mdoAdminDetails = (Map<String, Object>) request.getRequest();
+                String mdoAdminUserId = (String) mdoAdminDetails.get(Constants.USER_ID);
+                String reportExpiryDateRequest = (String) mdoAdminDetails.get(Constants.REPORT_EXPIRY_DATE);
+                Map<String, String> headersValue = new HashMap<>();
+                headersValue.put(Constants.CONTENT_TYPE, Constants.APPLICATION_JSON);
+                String mdoLeaderUserId = accessTokenValidator.fetchUserIdFromAccessToken(authToken);
+                if (null == mdoLeaderUserId) {
+                    throw new HttpServerErrorException(HttpStatus.INTERNAL_SERVER_ERROR, "UserId Couldn't fetch from auth token");
                 }
-                if (rootOrgIds.size() != 1) {
-                    logger.error("Failed to grant access due to different org.");
-                    throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "Failed to grant access due to different org.");
-                }
-                StringBuilder uri = new StringBuilder(serverConfig.getSbUrl());
-                Map<String, Object> roleRead = (Map<String, Object>) outboundRequestHandlerService.fetchResult(String.valueOf(uri.append(serverConfig.getSbRoleRead()).append(mdoAdminUserId)));
-                List<Map<String, Object>> roles = (List<Map<String, Object>>) ((Map<String, Object>) roleRead.get(Constants.RESULT)).get(Constants.ROLES);
-                List<String> assignedRoles = new ArrayList<>();
-                for (Map<String, Object> roleMap : roles) {
-                    assignedRoles.add((String) roleMap.get(Constants.ROLE));
-                }
-                if (!assignedRoles.contains(Constants.MDO_ADMIN)) {
-                    throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "The Grantee doesn't have MDO_ADMIN role");
+                logger.info("MDO Leader User ID : {}", mdoLeaderUserId);
+                StringBuilder url = new StringBuilder(configuration.getLmsServiceHost()).append(configuration.getLmsUserSearchEndPoint());//call sunbird user table =>rootorg set against the incoming rqst body.
+                Map<String, Object> userSearchRequest = getSearchObject(Arrays.asList(mdoLeaderUserId, mdoAdminUserId));
+                Map<String, Object> searchProfileApiResp = outboundReqService.fetchResultUsingPost(url.toString(), userSearchRequest, headersValue);
```

Comment:
use the existing methods in userutilityserviceimpl -- getUserDetailsFromDB method can be used here... 
do not construct request and process the response again -- which is already done inside userutilityserviceimpl